File: snobot_sim/src/main/native/cpp/SnobotSim/SimulatorComponents/Gyro/AnalogGyroWrapper.cpp
/*
 * AnalogGyrWrapper.cpp
 *
 *  Created on: May 5, 2017
 *      Author: PJ
 */

#include "SnobotSim/SimulatorComponents/Gyro/AnalogGyroWrapper.h"
#include "MockData/AnalogGyroData.h"

AnalogGyroWrapper::AnalogGyroWrapper(const std::shared_ptr<AnalogSourceWrapper>& aAnalogWrapper):
    IGyroWrapper("Analog Gyro"),
    mAnalogWrapper(aAnalogWrapper),
	mAngle(0)
{


}

AnalogGyroWrapper::~AnalogGyroWrapper() {

}


void AnalogGyroWrapper::SetAngle(double aAngle)
{
	mAngle = aAngle;
    HALSIM_SetAnalogGyroAngle(mAnalogWrapper->GetHandle(), mAngle);
}

double AnalogGyroWrapper::GetAngle()
{
	return mAngle;
}

File: snobot_sim_java/java/snobot_sim_java/com/snobot/simulator/simulator_components/gyro/AnalogGyroWrapper.java
package com.snobot.simulator.simulator_components.gyro;

import com.snobot.simulator.module_wrapper.ASensorWrapper;

import edu.wpi.first.hal.sim.mockdata.AnalogGyroDataJNI;

public class AnalogGyroWrapper extends ASensorWrapper implements IGyroWrapper
{
    protected final int mPort;
    protected double mAngle;

    public AnalogGyroWrapper(int aPort, String aName)
    {
        super(aName);
        mPort = aPort;
    }

    public void setAngle(double aAngle)
    {
        boolean isUpdate = mAngle != aAngle;
        mAngle = aAngle;
        if (isUpdate)
        {
            AnalogGyroDataJNI.setAngle(mPort, aAngle);
        }
    }

    public double getAngle()
    {
        return mAngle;
    }
}
